scvi/data/_preprocessing.py

```diff
@@ -452,3 +452,32 @@ def add_dna_sequence(
     sequence_df = pd.concat(seq_dfs, axis=0).loc[adata.var_names]
     adata.varm[sequence_varm_key] = sequence_df
     adata.varm[code_varm_key] = sequence_df.applymap(_dna_to_code)
+
+
+def reads_to_fragments(
+    adata: anndata.AnnData,
+    read_layer: Optional[str] = None,
+    fragment_layer: Optional[str] = "fragments",
+) -> None:
+    """Convert scATAC-seq read counts to appoximate fragment counts.
+
+    Parameters
+    ----------
+    adata
+        AnnData object that contains read counts.
+    read_layer
+        Key in`.layer` that the read counts are stored in.
+    fragment_layer
+        Key in`.layer` that the fragment counts will be stored in.
+
+    Returns
+    -------
+    Adds layer with fragment counts in `.layers[fragment_layer]`.
+    """
+    if read_layer:
+        data = np.ceil(adata.layers[read_layer].data / 2)
+    else:
+        data = np.ceil(adata.X.data / 2)
+
+    adata.layers[fragment_layer] = adata.X.copy()
+    adata.layers[fragment_layer].data = data
```

Comment:
```suggestion
    data = adata.layers[read_layer].data if read_layer else adata.X.data
    adata.layers[fragment_layer] =  adata.layers[read_layer].copy() is read_layer else adata.X.copy()
    adata.layers[fragment_layer].data = np.ceil(adata.layers[fragment_layer].data / 2)

```

Reply:
Would be great to also add a check with https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.issparse.html as this will give an uninformative error if the data is not in a sparse format.

Reply:
Thank you! I wrote it now so that is can handle both. Let me know if you'd prefer it differently.